### ISSUE TITLE
Success Criterion 2.5.8 - Touch Target Size (Minimum) - Level AA

### ISSUE THREAD

**Issue by JJdeGroot** [w3c/matf#10]
Discussion:

<details>
<summary>1 May 2024</summary>

[Source](https://www.w3.org/2024/05/01-matf-minutes.html#t04)

Joe: Pixel conversion: CSS pixel is the native density for the platform.

JJ: We should align with WCAG2ICT

<Mick> I asked this question recently in the MATF slack channel. The consensus was the same - to see 24 pixels being equivalent to whatever the appropriate unit used for other platforms - for iOS this is 24 points and for Android this is 24 density-independent pixels. This is an approach Deque follows as well I believe.

AA: should we just use mm on the screen, as specified by NN?

<Devanshu> WCAG2ICT notes: w3c/wcag2ict#293 (comment)

JJ: Android can modify density, so perhaps mm can be difficult

Joe: Can't verify, but it's consistent across devices and settings

<Mick> Would points or density-independent pixels be more helpful to developement teams then using mm?

quintinb: Users can define what they want the physical size to be, the separation is intentional

julian: Should we just follow the platform guidelines? Fingertips are variable in size and not precise. Shouldn't we be advocating for large touch targets?

Rachele: Shouldn't we just reference the AAA requirement?

Joe confirms

JJ: (AA) Spacing can be 24 from another target and then size can be anything

Sorry that was a reference to WCAG_AA not AA the user

<Mick> As an observation - we tried previously pushing Apple and Android sizes to our teams and got so much pushback from designers about it being too big.

<AA> I will make sure to add my name as AAK, to avoid confusion quintinb

ty AAK

quintinb: we also need to be considerate that Google does allow for 32x32dp in seemingly certain cases

JJ: Relying on public documentation can be problematic as the internal algo doesn't necessarily follow the 48x48 rule

https://github.com/google/Accessibility-Test-Framework-for-Android/blob/c65cab02b2a845c29c3da100d6adefd345a144e3/src/main/java/com/google/android/apps/common/testing/accessibility/framework/checks/TouchTargetSizeCheck.java#L164

JJ: We may have to figure out what Google's math is - maybe someone can look later

JJ A lot of people asking how to check size

quintinb: ADB can be used to extract the ally tree and the density, which can calculate the dp size

<Mick> A physical ruler or sending screenshots and using likes of photoshop. Neither massively accurate but have struggled to find other methods when not having access to code in audits.

JJ: but we need to be understanding of what tools auditors need

Joe: Uses the scanner with larger requirements because it reports the details of what is wrong

<Mick> That's clever Joe_Humbert

agree - thanks Joe - I think I failed to understand previously

Mick has done an audit against 2.2

They have been sending screenshots and measuring with a ruler

</details>

**1. Pixel Conversion:**
- Joe: Highlighted the importance of CSS pixel alignment, emphasizing its native density for the platform.
- JJ: Proposed aligning with WCAG2ICT standards, which received agreement from the group.

**2. Unit Considerations:**
- Mick: Raised a question regarding the equivalence of 24 pixels to other platform units, suggesting using points or density-independent pixels for consistency. Also mentioned previous resistance from designers to larger sizes.

**3. Advocacy for Large Touch Targets:**
- julian: Advocated for following platform guidelines and emphasized advocating for larger touch targets for better accessibility.
- Mick: Highlighted the pushback faced from design teams regarding the implementation of larger sizes.

**4. Accessibility Compliance Checks:**
- quintinb: Suggested using ADB for extracting ally tree and density, while Joe recommended using a scanner tool for detailed reports on accessibility issues.

**5. Tools and Techniques for Audits:**
- Mick: Shared methods used for audits, including using physical rulers and screenshots for measurements.

**Comment by illai** [comment]
We probably can not refer this success criteria without taking into consideration it's conflict with

**Comment by AlainVagner** [comment]
A few interesting stuff I have found reading closed issues on the WCAG repository about 2.5.8:
- we need to keep in mind the different purposes of 2.5.8 (AA) and 2.5.5 (AAA): ["with 2.5.8 the aim is to prevent mistaken clicks/taps, but with 2.5.5 the aim is correct click/taps."](https://github.com/w3c/wcag/issues/3714#issuecomment-1966167765)
- the value of 24px is coming from a compromise between what could benefit the users and the reality of websites with not exclusively touch interfaces ([source](https://github.com/w3c/wcag/issues/1831#issuecomment-850843764))
- about the differences between pt, dp, and css pixels: [ "For all intents and purposes, when evaluating native apps on Android/iOS, you basically just read "CSS px" as meaning dp for Android, point for iOS"](https://github.com/w3c/wcag/issues/3730#issuecomment-1982968344)

**Comment by jha11y** [comment]
<details>
<summary>Notes from 25 September, 2024 Meeting:</summary>

[Meeting Minutes 25 September](https://www.w3.org/2024/09/25-matf-minutes.html#t03)

quintinb: I think this is actually large considering there's no "unit standard" - Illai's MCAG does a great job at that

Jamie: +1 Illai

Illai: Both Android and iOS recommended sizes conflict with web - adopting 24x24 isn't sufficient

Google also has different sizes based on where elements exist on the screen: https://github.com/google/Accessibility-Test-Framework-for-Android/blob/c65cab02b2a845c29c3da100d6adefd345a144e3/src/main/java/com/google/android/apps/common/testing/accessibility/framework/checks/TouchTargetSizeCheck.java#L161

Joe_Humbert: we could add a strongly worded note that 24x24 is too small practically for human fingers

quintinb: How does this apply to links in mobile? (Just a point of interest ask)

Jamie: 1. in testing context, folks don't have access to the exact numbers (@Joe has a solution - Google Ally scanner (Quintin adds the Android Debug Bridge 'uiautomator dump')) 2. In practice this is most an exception because of spacing

https://m2.material.io/design/layout/pixel-density.html#pixel-density-on-the-web "When designing for the web, replace dp with px (for pixel)."

Joe_Humbert: we could refer to the bounding box for the hit target size

quintinb: +1 Joe_Humbert

Illai: +1 Joe_Humbert

ACTION: Add a note to 2.5.8 about a larger sizing requirement looking at the AAA requirement

</details>

**Comment by jamieherrera** [comment]
Pretty much @AlainVagner I'd agree on those points. Do you? The 2.5.8 exceptions exempt most scenarios for testing.

**Comment by jamieherrera** [comment]
@JJdeGroot could we maybe come back to this SC after some of the more well-worn A and AA SC have been discussed so that we can get through most SC in 2024? 

This SC would need some additional documentation that differs from the "Understanding 2.5.8" documentation, for mobile apps, though I guess that could be a later version

**Comment by AlainVagner** [comment]
@jamieherrera yes I do :) Just wondering if we add a note on 2.5.8 promoting a bigger touch target size on mobile, like 44 css px (or pt or dp), maybe we should also consider what would be the enhanced version of this SC, as there will probably be some impact on 2.5.5.

**Comment by julianmka** [comment]
Proposed note to address how small this SC would allow touch targets to be: 
>Twenty-four pixels is very small for targets on touch-based mobile devices. It is recommended that app makers use Success Criterion 2.5.5 Target Size (Enhanced) instead, making target sizes at least 44x44 pixels/DP/points.

Thoughts?

**Comment by JJdeGroot** [comment]
Discussed in yesterday's meeting.

<details>
<summary>9 October 2024</summary>

[Source](https://www.w3.org/2024/10/09-matf-minutes.html#d54b)

JJ: first discussion was 1 May 2024

JJ shares Issue from GitHub

JJ: Apple and Android have requirement for larger touch targets

JJ: explaining spacing exception in 2.5.8

JJ: continue discussion?

Detlev: go beyond WCAG (target size)?

JJ: stay in line with WCAG2ICT but some notes for best practices could be added

JJ: such as recommending larger sizes, reference developer documentation

JJ: Question was whether we better align with WCAG or WCAG2ICT -- the latter was preferred

Julian: Tell people to choose whatever is the more accessible (eg. decide between WCAG requirement and Android/iOS recommendations).
… Android and iOS are closed platforms where WCAG has no clout - that'S why we could instruct user sdto chosse the better one

JJ: could be trouble to link directly to Apple and Google

Joe: we should not use could language to inform people what they *should* be doing
… could is not used
… we could say it is too small for touch and one could point to triple A guideline without spacing exception

Jamie: as to adding Apple and Google resources: WCAG resources do include references to these
… "for information purposes only"
… not endorsed, but available - we could do it as well
… we are talking about AA 2.5.8 - we can point to AAA as best practice - should not linger longer, this is the AA standard

JJ: most people felt it can be applied 'as is'

JJ: comfortable with accepting as such and add a note to preferability of larger sizes?

<JJ> Add note to 2.5.8 to point people towards resources which recommend larger touch target sizes (such as WCAG 2.5.5, Apple, Google)?

<julianmka> +1

<Karla> +1

<Detlev> +1

<jeroen> +1

<Jamie> a+

<Joe_Humbert> +1

<Jamie> +1

<AlainVagner> +1

<Carolina> +1

Jamie: We would not be using the term CSS-pixels - replace with something else

JJ: may be density pixels or other units that make sense

ACTION: Add note to 2.5.8 to point people towards resources which recommend larger touch target sizes (such as WCAG 2.5.5, Apple, Google)

ACTION: Replace CSS Pixel unit with appropriate unit on mobile such as Density Pixel

JJ: will set up draft version of what guidance would look like
</details>

**Comment by JJdeGroot** [comment]
> Proposed note to address how small this SC would allow touch targets to be:
> 
> > Twenty-four pixels is very small for targets on touch-based mobile devices. It is recommended that app makers use Success Criterion 2.5.5 Target Size (Enhanced) instead, making target sizes at least 44x44 pixels/DP/points.
> 
> Thoughts?

To bring it a bit more in line with WCAG(2ICT) language:

> Targets with a size of 24 by 24 points may be too small for users on touch-based mobile devices. It is recommended to follow [Success Criterion 2.5.5 Target Size (Enhanced)](https://www.w3.org/TR/WCAG22/#target-size-enhanced), which recommends that target sizes should be at least 44 by 44 points."

**Comment by julianmka** [comment]
Much better, thanks @JJdeGroot! I agree that we should link to WCAG.

Once we have our own interpretation for 2.5.5, I wonder if we might even replace the guidance for this SC with something that amounts to "follow 2.5.8 because it's written with touch interfaces in mind." (Or perhaps a more strongly worded note?)

**Comment by JJdeGroot** [comment]
I think that might be changing the SC too much @julianmka 

Also, 2.5.5 is missing the 'Spacing' exception which has some downsides for some common mobile interface design patterns.

**Comment by julianmka** [comment]
Good points. And we do know, of course, that Google and Apple don't always follow their own guidance for touch target size, which almost always comes up in conversations about this SC. Under 2.5.8, though, most of the instances where touch targets are smaller than 44x44 would pass.

**Comment by dotjay** [comment]
I have opened issue https://github.com/w3c/matf/issues/84 for "Cross-platform Frameworks".

I have also posted in https://github.com/w3c/matf/issues/11#issuecomment-2621464584 about defining "screen" and "view" as a mobile-relevant equivalent to "web page", and suggesting "native mobile app" as a subset of "software".

**Comment by JJdeGroot** [comment]
Based on discussion above, added label drafting because it seems we are ready to make a draft proposal for applying this SC to mobile applications.

I recently wrote documentation about manually calculating density-independent points on Android and iOS.

**It involves various steps:**

1. Find display resolution (e.g. 720 width x 1600 height)
2. Find display size (e.g. 6.5 inches)
3. Calculate scaling
    - Android DPI: `√(width² + height²) ÷ size` [(DPI formula)](https://developer.android.com/training/multiscreen/screendensities#dips-pels) 
    - iOS Scale: [Scale reference](https://www.ios-resolution.com/)
4. Determine element size in pixels (e.g. 100 width x 100 height, determined by measuring screenshot)
5. Calculate element size in density-independent points
    - Android DP: `Pixels ÷ (DPI ÷ 160)` [(DP formula)](https://developer.android.com/training/multiscreen/screendensities#dips-pels) 
    - iOS Points: `Pixels ÷ Scale` [Scale formula](https://developer.apple.com/documentation/uikit/uiscreen/scale)
7. Determine if target size is sufficient, e.g. at least 24x24 DP or Points
8. If target size is insufficient, check for exceptions:
   - Check if spacing is sufficient, e.g. at least 24 DP or Points spacing
     - Follow steps 1-7 for each nearby target to calculate X and Y coordinates in density-independent points
   - Check if the same function is available via another, larger control on the same screen.
   - Check if the target is inline, such as a link or icon within a block of text.
   - Check if the target size is defined by the platform (e.g. native iOS or Android components) and not modified by the developer.
   - Check if a smaller size is essential or legally required.

Steps 1-3 only need to be repeated once per device.
Step 4-8 needs to be repeated for each target on every screen that's part of the scope.

**Comment by JJdeGroot** [comment]
The screenshot below shows the results of following the steps above. The spacing exception is the most problematic one, without automated testing you will need to draw circles that represent 24 dp/pt on top of a screenshot represented in pixels.

![Five targets are positioned in an app on an Android device. For each target, the width, height and spacing are calculated to determine whether they pass or fail the target size requirements.](https://github.com/user-attachments/assets/a621fc23-cc32-489f-a9bd-d7f6c60a3f82)

**Comment by shoobe01** [comment]
_Keeping as notes here vs trying my hand at editing the draft (partly as I am not confident I know how to do that properly), a stab at changes needed from my research and a bit of discussions on 8 Oct '25. I have for now selected 46 DIP as a middle between iOS and Android that is both 1) appx at the minimum size for center of screen from my research, and 2) is not any of the current OS standards, so we do not imply favoritism:_

**Touch Target Size (Minimum) (Level AA)** [Adding "touch" the title]
_[I also have thoughts on the In Brief section. Terms, "click" vs "tap," and some people when these are AA so baseline for most people to use products at all]_

**In Brief**
Goal
- Make interaction with touchscreens easier, more accurate, and faster.

What to do
- Ensure all selectable items meet a minimum size and have sufficient spacing between adjacent selectable items. 

Why it's important
- Excessively small items, especially very close, cannot be accurately clicked. This is particularly difficult for certain situations and for certain users with physical impairments.

**Success Criterion (SC)**
The size of the target for touch pointer inputs is at least 46 device pixels, except when: 

Edges
Targets near the edges of the display screen should be larger to account for the reduced touch accuracy people have as they move away from the center. Along the top and bottom edges the minimum size should be 78 DIP. 
_[78 is my safe R95 minimum for bottom edge, top edge is only fractionally more accurate]
[My ideal would be to "slowly increase size to 78 px as the edges are approached" or similar phrasing but that's not as easily testable so I chose to give a hard region and limit, should solve /most/ issues, much better than where we stand now!]_

Spacing
Undersized targets (those less than 46 DIP, or less than 78 DIP along the top and bottom edges) are positioned so that if a circle of 46 DIP (or 78 DIP) is centered anywhere within a target, the circle does not intersect another target, or the circle for another undersized target;
_[Note I removed "bounding box" in favor of "target" with additional details, have switched from a square as the testable item to a circle for human factors/statistical truth reasons, tweaked some other language, and explained more since it almost is a test protocol already; "centered" may need more work...]_

Equivalent... [unchanged]

Inline
~~The target is in a sentence or its size is otherwise constrained by the line-height of non-target text~~ **can we simply remove exceptions or do we need to un-exception it, explicitly say why it isn't a good idea for our domain (as explained in previous comment)?**

User Agent Control
_Updated since originally posted:_
Elements that may appear but are on a layer outside the current page or application, such as operating system controls. The size of the target is determined by the [user agent](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html#dfn-user-agent) and cannot be modified within this layer;

Essential... [unchanged]

Note 1... [unchanged]

Note 2... [unchanged]

Note 3 **Unless we cannot add a back door definition and need this to be a complete defined term**
Device pixel is the device independent pixel measurement unit specific to the operating system, Android DPs and iOS points. Defining the size is web pixels is prone to misinterpretation for mobile OS developers. 
**[Device pixel subtask discussion](https://github.com/w3c/matf/issues/157)]**

NOTE for us: I may have made a confusing statement re the scale issue for tablets while on the call. For touch sizing this actually works fine. For text sizes is where this is an issue (phone density, but used further away so need larger sizes). No exception needed.

NOTE for us: I read through but didn't document the Intent variations yet as that's a lot of stuff, and it will have to follow the actual SC decisions so this can be first. Aside from updating numbers and units, there are a few things that need to be looked at, e.g. "People with large fingers, or who are operating the device with only a part of their finger or knuckle." This is simply untrue; finger size has NO correlation to touch accuracy, nor does age, region, experience, etc. It is a holdover from the original 1970s ISO 9241-410 (_nee_ 9241-9) for which there is no evidence (appears to be an artifact of technology used at the time so limits of what testing they did perform).

**Comment by TanyaVanWorkum** [comment]
This exception might need more explanation (currently blocked by the definition of the [user agent ](https://github.com/w3c/matf/issues/225)and [“not modified by the author”](https://github.com/w3c/matf/issues/230)).

> User Agent Control
>     The size of the target is determined by the [**user agent**](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html#dfn-user-agent) and is **not modified by the author**;

**Comment by shoobe01** [comment]
Note that the 9 October '25 comment on [#155](https://github.com/w3c/matf/issues/155#issuecomment-3386057790) is a nice, brief, coordinated list of my thinking on sizes for AA and AAA together; even if we're only publishing one at a time, good to consider it all together to assure a consistent logic. 

Otherwise, scroll up and I have a good set of notes that correlated to the [implied: "mouse"] Target Size SC 2.5.8 in 2.2. See [the later comments on 8 October](https://github.com/w3c/matf/issues/10#issuecomment-3382933093).

**Comment by shoobe01** [comment]
Re: 24 px acceptable for touch,  this is the cited research on WCAG 2.2 for 2.5.8:  
https://dl.acm.org/doi/10.1145/1152215.1152260

I am aware of it. A bit poorly designed as a study, likely by being early in mindspace of touch, assuming everyone uses one thumb (not true, so forced behavior) and not addressing location on screen. For the center-of-screen where clicks were, it complies with the sizing other studies found. 7.7 mm is 43 px. 
Unless some strange scaling issue occurred, not following how this supports 24 px on touch. 


Here is the bulk of the studies I have included before when I published these, and the notes that accompanied them to say what's interesting about them  (I have my own summary articles including in ACM Interactions, and a book... so can always reference: myself!) 

- Bérard, François. “Measuring the Linear and Rotational User Precision in Touch Pointing.” Proceedings of the 2012 ACM International Conference on Interactive Tabletops and Surfaces. New York: ACM, 2012. A detailed, well-designed exploration of separating physical pointing limits from other aspects of touch. It turns out that people can control their finger positions to an accuracy of around 0.1 mm. However, except in cases where a user interface provides enhancements—such as the edit magnifier in iOS—this is irrelevant, and we must consider all aspects of perception that result in pointing accuracy.
- Bergstrom-Lehtovirta, Joanna, and Antti Oulasvirta. “Modeling the Functional Area of the Thumb on Mobile Touchscreen Surfaces.” Proceedings of the 32nd Annual ACM Conference on Human Factors in Computing Systems (CHI ’14). New York: ACM, May 2014.
- Bragdon, Andrew, et al. “Gesture Play: Motivating Online Gesture Learning with Fun, Positive Reinforcement and Physical Metaphors.” Proceedings of ITS ’10 ACM International Conference on Interactive Tabletops and Surfaces. Saarbrücken, Germany: ACM, 2010. Bragdon is one of the few researchers who has really carefully, scientifically explored gestures. This research is still foundational, so we can currently draw few conclusions that are applicable to design, but it’s interesting stuff.
- Gilbert, Juan E., Aqueasha M. Martin, Gregory Rogers, Jerome McClendon, and Josh Ekandem. “Hey, That’s Not Who I Voted For! A Study on Touchscreen Ballot Design.” Interactions, November 2012. Retrieved March 28, 2017. This article conflates target and interference, and all tested designs have buttons that are hard against each other—so watch out for that. However, it includes interesting information on indicator and visual-target focus and discusses labels attracting taps and checkmarks to indicate selection.
- Henze, Niels, Enrico Rukzio, and Susanne Boll. “100,000,000 Taps: Analysis and Improvement of Touch Performance in the Large.” Proceedings of the 13th International Conference on Human Computer Interaction with Mobile Devices and Services. New York: ACM, 2011. This is my favorite research on touch that I didn’t do myself. They got several million somewhat-qualified clicks from a game on Android, so the research occurred outside the lab and has insane scale. “Below 15 mm, the error rate dramatically increases and jumps to over 40% for targets smaller than 8 mm.” There are nice charts on error rates by position, showing that the edges of the screen are much worse, pretty symmetrically.
- Lee, Seungyon, and Shumin Zhai. “The Performance of Touch Screen Soft Buttons.” Proceedings of the SIGCHI Conference on Human Factors in Computing Systems. New York: ACM, 2009. 10-millimeter buttons resulted in 98 to 99.4% accuracy. Reducing button size caused users to switch from thumb to finger. iPhone keyboard buttons resulted in high accuracy (90%), but a typing speed that was a third slower; plus, there was a huge jump in the corrections that were necessary when typing on the narrow keyboard. I think they measured accuracy as the ability to enter the right value, not the user’s ability to hit the right target the first time, and therefore, missed critical interference issues.
- Ng, Alexander, Stephen A. Brewster, and John H. Williamson. “Investigating the Effects of Encumbrance on One- and Two-Handed Interactions with Mobile Devices.” Proceedings of the 32nd Annual ACM Conference on Human Factors in Computing Systems (CHI ’14). New York: ACM, May 2014. This is where I got the very interesting information on how much touch is affected by holding a bag in the other hand.
- Parhi, Pekka, Amy K. Karlson, and Benjamin B. Bederson. “Target Size Study for One-handed Thumb Use on Small Touchscreen Devices.” Proceedings of the 8th Conference on Human-Computer Interaction with Mobile Devices and Services. New York: ACM, 2006. Clear research findings showing that interference sizes are smallest at the center of the screen and notably worse at the corners. If you have access to the paper, its Figure 6 provides a simple explanation.  “[The] target size of 9.2 mm for discrete tasks and targets of 9.6 mm for serial tasks should be sufficiently large for one-handed thumb use on touchscreen-based handhelds.” They used CEP-R95, and there was no improvement when increasing target size to 11.5 mm. This paper conflates target size and interference for the most part, very often putting test items adjacent to each other. In Mobile Usability, quoted by Nielsen as recommending a target size of around 10 mm, which I can agree with.
- Park, Yong S., Sung H. Han, Jaehyun Park, and Youngseok Cho. “Touch Key Design for Target Selection on a Mobile Phone.” Proceedings of the 10th International Conference on Human Computer Interaction with Mobile Devices and Services. New York: ACM, 2008. Really interesting research because participants cradled the device and used their thumb to interact and the results plotted the accuracy rate by screen location. Size recommendations resulted from this study as well—specifically, that 10 millimeters is better than 7 millimeters, and 4 millimeters is sort of a disaster. The authors made special note that the bottom of the screen was hard to tap accurately.
- Perry, Keith B., and Juan P. Hourcade. “Evaluating One-Handed Thumb Tapping on Mobile Touchscreen Devices.” Proceedings of Graphics Interface 2008. New York: ACM, 2008. Evaluated users both walking and standing, and all held the device with one hand and tapped with the thumb.
- Plait, Phil. “[Resolving the iPhone Resolution](http://blogs.discovermagazine.com/badastronomy/2010/06/10/resolving-the-iphone-resolution/#.WOQzjRLyvUo).” Discover, October 17, 2014. Retrieved March 28, 2017. In the guise of adding science to the hype around the original release of the Apple Retina display on iPhone, Phil Plait, in his Bad Astronomy column, published an easy-to-read discussion of the math behind angular resolution and why it matters more than absolute size.
- Schildbach, Bastian, and Enrico Rukzio “Investigating Selection and Reading Performance on a Mobile Phone While Walking.” Proceedings of the 12th International Conference on Human Computer Interaction with Mobile Devices and Services. New York: ACM, 2010. “Whilst performance decreases, cognitive load increases significantly when reading and selecting targets when walking. Furthermore, the results show that the negative effect regarding target selection can be compensated by increasing the target size, but the text reading task did not yield better performance results for a larger text size due to the increased demand for scrolling.” Here are this study’s results: 6.74-millimeter buttons had an error rate of up to 23%; buttons of about 10 millimeters were much better; and 20-millimeter buttons solve almost all problems, even when the user is walking. Small buttons make people slow down due to cognitive loading. While small, 6-point text results in no performance degradation when the user is walking, some users perceive it as being worse. There is a middle ground due to the need to scroll. Text sizes that are too large are also bad.
- Schmidt, Mikkel. “[Mobile UI Ergonomics: How Hard Is It Really to Tap Different Areas of Your Phone Interface?](https://medium.com/@mibosc/mobile-ui-ergonomics-how-hard-is-it-really-to-tap-different-areas-of-your-phone-interface-bb043d409af8)” Medium, October 17, 2014. Retrieved March 28, 2017.
- Sesto, Mary E., Curtis B. Irwin, Karen B. Chen, Amrish O. Chourasia, and Douglas A. Wiegmann. “Effect of Touch Screen Button Size and Spacing on Touch Characteristics of Users With and Without Disabilities.” Human Factors: The Journal of the Human Factors and Ergonomics Society, June 2012, Volume 54, Number 3. The study found that 20-millimeter buttons worked best and larger spacing between them didn’t help, but they measured the force used, not targeting accuracy—apparently because of a special need of the disabled cohort. While these findings are not directly applicable, the 20-millimeter size is reasonable when accounting for inaccuracy due to motor-function issues.
- Stack Overflow. “[Android Minimum Button Height: 48dp = 9mm?](http://stackoverflow.com/questions/10112780/android-minimum-button-height-48dp-9mm)” Stack Overflow. Retrieved March 28, 2017. A discussion of the scale issues that are associated with using Android dp for physical sizes.
- Wroblewski, Luke. “[Responsive Navigation: Optimizing for Touch Across Devices](http://www.lukew.com/ff/entry.asp?1649).” LukeW Ideation + Design, November 2, 2012. Retrieved March 28, 2017.
- Xu, Wenchang, et. al. “Digging Unintentional Displacement for One-handed Thumb Use on Touchscreen-Based Mobile Devices.” Proceedings of the 14th International Conference on Human-Computer Interaction with Mobile Devices and Services. New York: ACM, 2012. Detailed discussion of the inaccuracies that liftoff—removing your finger from the screen—induces on mobile handsets, with some exploration of the biomechanics that are involved and how those inaccuracies change according to the position on the screen. Interesting notes about changes in accuracy over time, which will inform any gesture standards that may eventually emerge.

My summary of this research, and [my own primary research](https://www.uxmatters.com/mt/archives/2014/09/insights-on-switching-centering-and-gestures-for-touchscreens.php) (including some done in coordination with [the eLearning guild](https://shoobe01.blogspot.com/2014/11/making-mlearning-usable-how-we-use.html)) 

- People shift their grip on portable mobile devices, all the time. There is no obvious preferred method per person, it is mostly or all environmentally and task driven; cannot assume one-thumb use, for example. 
- The oft-repeated thumb sweep zones, that tell you all touches must be at the bottom are wrong. Not supported by any research. (I helped spread this, my old refuted diagrams are spread all the time much to my regret). 
- Devices are held closer to the eye the smaller they are. Total screen angular size is similar for all devices, from small phone to movie screen. Not important for touch, but critical for type sizes. 
- Larger devices are used more in landscape than smaller. Surprising everyone: no obvious applicability to touch but there it is. 
- Touch accuracy varies by position on the screen: 
  - 7 mm at the center – 39 px
  - 9 mm along the edges – 51 px
  - 12 mm in the corners – 61 px
- Speed of touch, and speed and accuracy of reading, correlates to the screen position; content is scrolled to the center naturally, etc. 
- Touch accuracy and preferences have no perceptible change with device size changes. As long as it's handheld, from smallest phone to large tablets, accuracy by relative position on the screen is the same. 
- Touch accuracy doesn't vary by region, language, age, sex or gender, experience level or any other data gathered. 
- Hold preference (and touch accuracy) has no clear relationship to stated user handedness. There are significantly more left-dominant touches than the generally accepted rate of left handed in the population. This is slightly confounding and needs to be investigated more. 
- Touch accuracy vary wildly based on user limitations (disability) such as tremor, and for temporary-disability such as holding something in your other hand, walking or riding a vehicle instead of being stationary, etc.

**Comment by JJdeGroot** [comment]
[Discussed in MATF meeting on 4 March 2026](https://www.w3.org/2026/03/04-matf-minutes.html#d4a8)

<details>
<summary>Minutes for 4 March 2026</summary>

~~~text
<JJ> [w3c/matf#10](https://github.com/w3c/matf/issues/10)

<pauljadam> Do we need to recommend or mention in a note that Apple asks for 44 and Android asks for 48 which is beyond WCAG minimums?

The scanner for Android: https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor

<pauljadam> The testing tools Android A11y Scanner and Xcode A11y Inspector do not account for spacing so you get false positives when using those tools.

<JJ> [w3c/matf#10 (comment)](https://github.com/w3c/matf/issues/10#issuecomment-3298969073)

<shoobe01> [w3c/matf#155 (comment)](https://github.com/w3c/matf/issues/155#issuecomment-3386057790)

shoobe01 this is meta-research about sizes, I'm very confident about this, and trying to provide a link at the bottom (in IRC) has a summary. Our decision making should include both 2.5.8 and 2.5.5

shoobe01 "interference size" deals with how big an exclusion zone you need

What do you mean keeping 24?

and why?

shoobe01disappointed by that - it's plausable for pointer driven systems. But this shouldn't be applied for touch. Touch target size is what it is. We need to completely revamp this because of the entirely different interaction

<pauljadam> Should be called "Target Size or Spacing"

Joe_Humbert we should probably include reasons why we aren't making sweeping changes. We may need to talk to the larger working group. How many people read the preamble, because we need to ensure that people understand why.

<JJ> Poll: Do you agree to increase the minimum target size requirements from 24 to a larger number because of touch target vs pointer target differences?

<shoobe01> +1

<quintinb> +1

<Joe_Humbert> shoobe01 this is the cited research on WCAG 2.2 for 2.5.8. https://dl.acm.org/doi/10.1145/1152215.1152260 I'm sure they reviewed other research, but I would have to ask

<tayef> +1

<Joe_Humbert> +1 but not sure of what number

<pauljadam> +1 we should make AAA moved to AA

<JatinV> 0

<pauljadam> the spacing part of AA is not even easily testable

<Joe_Humbert> +1 to what pauljadam said about spacing

ACTION: 5x +1 and 1x 0 for increasing 2.5.8 minimum size requirements

Agree pauljadam - how are wesupposed to test a gap if everything is in odd units like dp?

<pauljadam> dp pt px it's just 44 x 44

<pauljadam> iOS 44 android 48

(I was talking about the gaps)

<pauljadam> gap is not part of the test for the AAA 44 x 44 requirement

I know, that's why I agree with you :)

<pauljadam> lol :)

<pauljadam> sorry I thought you were asking me a question lol 😆

oh right - apologies - yeah I can see how it can read like that! It was a lament

<pauljadam> yes I've written test case steps for testing target size AA and it's weird to write because the automated tools will show false positives and you have to tell the tester to manually look for spacing

ACTION: Create draft for 2.5.8 with larger size and notes with best practices for touch target size (vs pointer)

<pauljadam> people often ask how do they measure a target size on a mobile, they'll wonder if they need to pull out a ruler :)
~~~
</details>

**Poll:** Do you agree to increase the minimum target size requirements from 24 to a larger number because of touch target vs pointer target differences?

**Results:** 5x +1 and 1x 0 for increasing 2.5.8 minimum size requirements

- ACTION: Create draft for 2.5.8 with larger size and notes with best practices for touch target size (vs pointer)

**Comment by detlevhfischer** [comment]
> Poll: Do you agree to increase the minimum target size requirements from 24 to a larger number because of touch target vs pointer target differences?
> Results: 5x +1 and 1x 0 for increasing 2.5.8 minimum size requirements

I wasn't in the meeting but here is my belated vote: -1

I would suggest **not to deviate from WCAG2ICT** (which is also what the normative baseline we will (have to) use in Europe, EN 301 549, will refer to ) and to couch all additions or deviations as recommendations --  **OR to decide, and make it crystal clear to our audience, that MATF/WCAG2Mobile consciously deviates from WCAG2ICT and the European norm** and happily goes beyond it.

**Comment by shoobe01** [comment]
**Baseline** since the discussion is getting long:

This is my last edited suggested wording for no. 10 (with tons of notes about why): 
https://github.com/w3c/matf/issues/10#issuecomment-3382933093

This comment in no. 155 is a summary of my plan for AA and AAA to make sure that we don't have any conflicts defining one now, and the other later: https://github.com/w3c/matf/issues/155#issuecomment-3386057790

----------

The more I thought about the discussion of increasing size for touch I have thought this is not a platform thing, this is a user interaction thing. A mode of interaction. (ETA: I think this user-modality thinking also complies well with the common approach these have, thinking of assistive tech or a11y interfaces, vs kb/mouse/screen display)

And that means we need for both mouse/trackpad/etc standards to be preserved, **and** to have touchscreen standards. Even on mobile: 

Imagine an iPad app designed explicitly for use with a keyboard and trackpad attached (this is increasingly common hardware; many businesses are issuing that to supplement or replace laptops). It's a mobile device so in the MATF scope, still but it would need to comply only with the existing 24 pixel standard because it assumes mouse driven (and the opposite could occur on Windows, if touch is assumed...). 

Other apps that assume (or support without switching) touch would need a new larger size like the 46/78px I proposed. 

So maybe a tangential solution to "make it crystal clear...that MATF...consciously deviates" to have **both as suitable SCs**, depending on the interaction device expected. Does this mean: 
- a note to add this to 2.5.8
- or
- an all new SC for "Touch Target Size"?  

ETA: If a note on existing SC, can we just add one with different rules to the Exceptions (and add supporting information to the Understanding page)? 

ISSUES or OPEN: 
- Units: The existing SC (mouse) is in web px. We have been working towards all size defs for MATF being in native device independent pixels. Combining the two specs in one SC risks a lot of confusion and/or a lot of work. Show both sizes in web px AND dip? 
- Are we hard-limited in scope to mobile apps? It's a user mode not a platform, so touch regardless of web or app needs these sizes; possible to have that happen if we are allowed to change/add-to the existing SC.

**Comment by JJdeGroot** [comment]
[Discussed in MATF meeting on 1 April 2026](https://www.w3.org/2026/04/01-matf-minutes.html#45b4)

<details>
<summary>Minutes</summary>

~~~text
JJ: had a poll where most of group agreed that we should increase minimum target sizes from 24 to larger

JJ: Detlev voted no because this would not harmonize with other guidance

JJ: wondering if it's a good approach to change the normative guidance ?

shoobe01: added research as a comment to the target size ticket

<JJ> shoobe: research dump [w3c/matf#10 (comment)](https://github.com/w3c/matf/issues/10#issuecomment-3997891523)

shoobe01: research discusses pitfalls like scaling

JJ: in touch with folks working on WCAG 3 and asking us to do a review of WCAG 3

JJ: we may be able to have an impact on WCAG 3 and their related target size requirements

JJ: 5 years maybe until WCAG 3 finished but we could have a big impact since WCAG standard last so long

JJ: JJ will review WCAG3 and bring ideas to the group about diving work, maybe in google doc or GitHub issues and then bring that back to WCAG3 task force and have a positive impact there

JJ: in WCAG 2 mobile make drafts and get feedback and after our second draft is up for review
~~~
</details>

**Comment by shoobe01** [comment]
Current plan (which I will work on): 
- Add a brief note to 2.5.8, with a link to bullet below
- Create a Statement about all the suggested sizes and why (create a Google Doc to share with TF)
  - Intro: touch and other direct-select methods are very different from proxy select methods like mouse, trackpad, trackball... 
  - baseline on human factors truths from [bottom of this](https://github.com/w3c/matf/issues/10#issuecomment-3997891523), do exceptions about temporary and permanent disability, etc. etc.
  - reference back to 2.5.5/2.5.8, and give explicit recommended sizes for AA and AAA; give sizes in DIP AND web PX
  - explain interference vs touch 
  - explain units, how DIP is not web PX but we give both because humans, must work on any platform
  - Touch on research used to find this, start by pasting [these in](https://github.com/w3c/matf/issues/10#issuecomment-3997891523) for reference
  -  Make SURE to indicate platform standards (ideally without naming Google/Apple) suggest larger sizes, which vary but are similar to our standard
  - Consider statement of things that do not apply, but do keep it to accuracy
    - The old ISO standards of how accuracy and finger sizers have no relevance
    - How things like Fitts' law don't apply,... maybe? Lots of tactics are not directly size related though... 

AGWG would have to approve this at all, much less in detail. 

Use this as a model for the format perhaps: https://www.w3.org/TR/ethical-web-principles/

**Comment by shoobe01** [comment]
Google Doc (WIP):
https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?usp=sharing

(No authorization required... do we want to restrict access???)

**Comment by shoobe01** [comment]
Alternative is a Note, like this one: https://www.w3.org/TR/web-user-agents/

**Comment by shoobe01** [comment]
Next draft for SC update and link to our external Statement/Note/Something document to get into details. Only change here is to add a third note. Full text of the SC is here for context only. 

**Success Criterion 2.5.8 Target Size (Minimum)**
The size of the [target](https://www.w3.org/TR/WCAG22/#dfn-targets) for [pointer inputs](https://www.w3.org/TR/WCAG22/#dfn-pointer-inputs) is at least 24 by 24 [CSS pixels](https://www.w3.org/TR/WCAG22/#dfn-css-pixels), except when:

**Spacing**
Undersized targets (those less than 24 by 24 CSS pixels) are positioned so that if a 24 CSS pixel diameter circle is centered on the [bounding box](https://www.w3.org/TR/WCAG22/#dfn-bounding-boxes) of each, the circles do not intersect another target or the circle for another undersized target;
**Equivalent**
The function can be achieved through a different control on the same page that meets this criterion;
**Inline**
The target is in a sentence or its size is otherwise constrained by the line-height of non-target text;
**User Agent Control**
The size of the target is determined by the [user agent](https://www.w3.org/TR/WCAG22/#dfn-user-agents) and is not modified by the author;
**Essential**
A particular [presentation](https://www.w3.org/TR/WCAG22/#dfn-presentation) of the target is [essential](https://www.w3.org/TR/WCAG22/#dfn-essential) or is legally required for the information being conveyed.

**Note 1**
Targets that allow for values to be selected spatially based on position within the target are considered one target for the purpose of the success criterion. Examples include sliders, color pickers displaying a gradient of colors, or editable areas where you position the cursor.

**Note 2**
For inline targets the line-height should be interpreted as perpendicular to the flow of text. For example, in a language displayed vertically, the line-height would be horizontal.

**Note 3**
User interaction with touchscreens, especially on small mobile devices, is different than that with cursor-based systems such as mouse and trackpad. It requires different and larger sizes, and for the sizes to be specified in native units as well as CSS pixels. Wording for link??? https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?usp=sharing

**Comment by JJdeGroot** [comment]
[Discussed in MATF meeting on 3 June 2026](https://www.w3.org/2026/06/03-matf-minutes.html#5e8b)

<details>
<summary>Minutes</summary>

~~~text
<JJ> https://www.w3.org/standards/types/#x2-5-3-statements

<JJ> https://www.w3.org/TR/?filter-tr-name=&status%5B%5D=stmt\

shoobe01 we have a lot of specific problems like units, which will create confusion which is hard. Can we make our own success criteria? Can we make our own statement so that people will find it? Do we put a link?

<JJ> https://www.w3.org/TR/ethical-web-principles/

shoobe01 is there a writing guide for these? Do we copy the existing ones?

Would these be normative?

shoobe01 I can talk about this for a long time - I doubt we want anything resembling the findings to be in a note like this. I'll start roughly organsing to see if we're on the right page

shoobe01 hopefully this get's absorbed into WCAG3

ryomtoob I was wondering if we had any thoughts about referencing apple and google's documentation? I was scrolling through the note, and we don't have them. Do we plan on factoring them in?

shoobe01 They don't provide scientific data or math - which is why we didn't

shoobe01 I would wonder if we would be allowed to do it? Maybe we can avoid brand names and roll it into guidance

ryomtoob Maybe something generic to reference "Owners of the systems"
~~~
</details>

Insight: we could publish a separate document with guidance on touch target size.

**Comment by JJdeGroot** [comment]
[Discussed in MATF meeting on 10 June 2026](https://www.w3.org/2026/06/10-matf-minutes.html#aa23)

<details>
<summary>Minutes</summary>

~~~text
Related minutes from last week: https://www.w3.org/2026/06/03-matf-minutes.html#5e8b

Tanya: We need to figure out the best format for the research done by shoobe01

Tanya: Kevin White (W3C Staff) and Hidde de Vries (W3C ex-staff) were asked if a "Statement" could be an appropriate document type for touch target size recommendations.

Tanya: Shows note about Web User Agents as example

https://www.w3.org/TR/web-user-agents/

Google Doc created by shoobe01: https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?usp=sharing

currently at 12 pages.

3 pages of references.

Tanya: we'll see how to move forward with this. Can we share it with others?

JJ: Will share it with Hidde to get feedback

<shoobe01> https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?usp=sharing

shoobe01: will continue finetuning.

Tanya: We need to figure out how to link to this draft document or note in the future from the Success Criterion itself

JJ: Suggest to link to Google Docs URL in the note for now. In the future we can update URL. If URL changes, we can put banner in Google Doc that mentions new URL.

shoobe01: asks about the note that is needed, what is the plan?

Tanya: confirms that we will now add note with 1 paragraph about reasons to use larger target size

JJ: This target size document can influence WCAG 3 in future. Unlikely WCAG2 will change
~~~
</details>

Document with consolidated research: https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?usp=sharing

**Comment by JJdeGroot** [comment]
@shoobe01  I have added your proposal to a branch and PR: https://github.com/w3c/matf/pull/291/changes

We do need to double check what we will substitute "user agent" with. For now, I used "application", but could also be "platform", or something else.

We also need to check the CSS Pixel unit, and if we want to substitute that with something like "density-independent unit".

**Comment by shoobe01** [comment]
I am a bit confused about this: 
> @shoobe01 I have added your proposal to a branch and PR: https://github.com/w3c/matf/pull/291/changes

Only Note 3 is changed (well, added) to the existing SC, so the coding of what's changed seems off. If it's right, someone walk me through this so I get my head on right. 

> We do need to double check what we will substitute "user agent" with. For now, I used "application", but could also be "platform", or something else.

Ah... I hadn't touched any existing SC content so I guess this would become a dependency on what we choose for layering and the redefining of UA in issue #63. As it stands, we'd just replace UA with "Application" or "Application layer." I rather like adding layer so it encompasses that several types of things can be there, and better anchors to whatever the output for #270 will be, presumably a definition we can inline link to. 

> We also need to check the CSS Pixel unit, and if we want to substitute that with something like "density-independent unit".

Much defined in the G Doc I created, where I have defined sizes in both DIP and CSS-px, based on what the output is rendered in. I used "native units" in note 3 here as I didn't want to start introducing other concepts, worried DIP even spelled out could be confusing and too long.

**Comment by shoobe01** [comment]
Still confused about what is up with the formatting of the commit (or whatever that is?) so edit here to address request to finalize content for new Note 3 per To Do from 24 June 2026: 

> **Success Criterion 2.5.8 Target Size (Minimum)**
> The size of the [target](https://www.w3.org/TR/WCAG22/#dfn-targets) for [pointer inputs](https://www.w3.org/TR/WCAG22/#dfn-pointer-inputs) is at least 24 by 24 [CSS pixels](https://www.w3.org/TR/WCAG22/#dfn-css-pixels), except when:

> **Spacing**
> Undersized targets (those less than 24 by 24 CSS pixels) are positioned so that if a 24 CSS pixel diameter circle is centered on the [bounding box](https://www.w3.org/TR/WCAG22/#dfn-bounding-boxes) of each, the circles do not intersect another target or the circle for another undersized target;
> **Equivalent**
> The function can be achieved through a different control on the same page that meets this criterion;
> **Inline**
> The target is in a sentence or its size is otherwise constrained by the line-height of non-target text;
> **User Agent Control**
> The size of the target is determined by the [user agent](https://www.w3.org/TR/WCAG22/#dfn-user-agents) and is not modified by the author;
> **Essential**
> A particular [presentation](https://www.w3.org/TR/WCAG22/#dfn-presentation) of the target is [essential](https://www.w3.org/TR/WCAG22/#dfn-essential) or is legally required for the information being conveyed.
> 

> **Note 1**
> Targets that allow for values to be selected spatially based on position within the target are considered one target for the purpose of the success criterion. Examples include sliders, color pickers displaying a gradient of colors, or editable areas where you position the cursor.
> 
> **Note 2**
> For inline targets the line-height should be interpreted as perpendicular to the flow of text. For example, in a language displayed vertically, the line-height would be horizontal.
> 

**Note 3**
User interaction with touchscreens, especially on portable mobile devices, is different than that with cursor-based systems such as mouse and trackpad. It requires different and larger touch target sizes, and for the sizes to be specified in native device independent units as well as in CSS pixels. Read the complete MATF Group Note Draft on [Touch Target Sizes](https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?usp=sharing) for detailed guidelines, explanation, and references.

**Comment by TanyaVanWorkum** [comment]
[Discussed in MATF meeting on 24 June 2026](https://www.w3.org/2026/06/24-matf-minutes.html4)

<details>
<summary>Minutes</summary>

~~~text
<Tanya>[ https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?tab=t.0#heading=h.cos5gmam7jdr](https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?tab=t.0#heading=h.cos5gmam7jdr)
Tanya: Steven research on touch target size shared
Tanya: target size says it should be 24 by 24 CSS pixels and we can't change these numbers in our note so we may need to make a new note that states info from Steven's research
<Tanya>[ w3c/matf#10 (comment)](https://github.com/w3c/matf/issues/10#issuecomment-4670758562)
Tanya: add comments in GitHub if you have any about the note or document
Tanya: we need to finalize these
Tanya: leave comments if you have any on the definitions so we can move forward with the final draft
~~~
</details>

* **ACTION**: Finalise Note 3.
* **ACTION**: Review the [Note on Target Sizes on Mobile Devices](https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?tab=t.0#heading=h.cos5gmam7jdr) and respond to Steven’s questions before the next meeting.

**Comment by jwlilly** [comment]
I think adopting a density-independent unit requirement is the right approach here. I agree with mapping the 24 CSS pixel threshold to its native device-specific equivalents (dp and pt).

To test how closely these units align in practice, I built a test app in React Native that allowed me to compare identical UI layouts rendered as Web, native Android, and native iOS.

Web reports this as 24x24 px. This is an direct inspection of the HTML running on-device:

<img width="1510" height="954" alt="Screenshot of the test app showing 24 x 24 px for a web page" src="https://github.com/user-attachments/assets/492dc406-18c5-44f7-b7e3-bd45ff9e1908" />

When inspecting the web version via Android's accessibility service layer, Chrome reports the bounding box as 26.29 x 25.90 dp. This variance likely comes from how the browser engine scales the viewport and factors in borders, padding, or fractional pixel rendering when sending coordinates to the OS accessibility tree.

<img width="1250" height="850" alt="Screenshot of android accessibility inspector showing the size of 26.29 x 25.90 dp for the same element" src="https://github.com/user-attachments/assets/750cef4d-a95c-4b4d-b6c0-0c3bd4811504" />

When the exact same component is rendered natively (not inside a web page), the Android accessibility tree reports the size as exactly 24 x 24 dp.

<img width="1248" height="851" alt="Screenshot of the natively rendered version in the android accessibility inspector reporting 24 x 24 dp" src="https://github.com/user-attachments/assets/92dbbc04-4133-45c8-9720-90f5100aefaf" />

iOS in Safari showing the same UI reports the element as 24 x 24 units. While Appium's inspector UI doesn't explicitly label the unit type, it is likely pulling directly from the iOS accessibility frame (UIAccessibilityElement.accessibilityFrame), which operates natively in Points (pt).

<img width="1712" height="944" alt="Screenshot of appium inspector showing the size as 24 x 24 units for the same element" src="https://github.com/user-attachments/assets/e6c3ede1-5cef-4206-a37d-1d26f9e5700c" />

**Comment by shoobe01** [comment]
> I think adopting a density-independent unit requirement is the right approach here. I agree with mapping the 24 CSS pixel threshold to its native device-specific equivalents (dp and pt).

So... add a DIP size to the 24 px (and 44 px?) sizes just for comparison or the few cases where mobile devices are used with mouse? See the end of 2.1 in the [Draft Note](https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?usp=sharing). 

> When inspecting the web version via Android's accessibility service layer, Chrome reports the bounding box as 26.29 x 25.90 dp. This variance likely comes from how the browser engine scales the viewport and factors in borders, padding, or fractional pixel rendering when sending coordinates to the OS accessibility tree.

Yup, you'll see if you do this on enough devices a stairstep effect (and even more if you measure the physical sizes), the scaling ratios do not line up neatly with actual screen densities, so they are just categorized. We need to make sure we're happy with a tolerance range for the sizes selected. 

This is one reason I propose a single DIP size for all mobile OSs; the difference between the two units is within their individual tolerance for scaling.

**Comment by jwlilly** [comment]
> So... add a DIP size to the 24 px (and 44 px?) sizes just for comparison or the few cases where mobile devices are used with mouse? See the end of 2.1 in the [Draft Note](https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?usp=sharing).

I think we only need to specify a density-independent unit size to the 24 px and 44 px values that corresponds to the equivalent size for the platform. For Android, that means 24 or 44 dp, and for iOS, it would be 24 or 44 pt. I would like to keep things platform-agnostic, but these are the primary two mobile platforms we have concrete evidence to support, and their density units are essentially equivalent to CSS pixels (at least based on my testing).

We should probably avoid using the general term DIP as an umbrella unit without giving some concrete platform examples (maybe in a definition) because iOS developers do not use that terminology in their code or IDEs. Stating dp for Android and pt for iOS gives developers the exact native language they need to audit their code accurately.

For the cases where mobile devices are being used with a mouse, I am not sure that I follow where that is going. The target size success criteria are intended to be input-agnostic. They evaluate the programmatic dimensions of the bounding box itself, regardless of whether the user is navigating via a mouse, a finger, or a switch. Because the rule applies identically across all pointers, I do not think we need to mention or define those usage scenarios differently.

We also need to be careful about introducing new numbers like 46 or 78 (from the 2.1 draft) into this note. Our main job is to map the published WCAG 2.2 standard to mobile frameworks. Adding new dimensional thresholds introduces non-standard compliance criteria that do not align with the original text of WCAG. If we publish numbers outside the official spec, we are going to cause a lot of confusion for compliance teams and auditors who rely on strict standard definitions.

**Comment by shoobe01** [comment]
Please read the Note in Google Docs I've written. It proposes a parallel set of sizes that are larger, specifically for touch based on a ton of research. 

>The target size success criteria are intended to be input-agnostic.
I presume that was the original intent, but I thought the point of the work of this TF is to point out and propose improvements when it demonstrably needs to change, due to platform or human interface differences. Touch is not very much like mouse/trackpad input. 

The Note is non-normative as yet, my understanding being that if it passes many other gates of approval it becomes a formal Statement and then can eventually be turned into (or modify existing) SCs, etc. The Note is a way to bring these fairly different standards in, more gently and with more broad group oversight and impact consideration than the original thoughts of adding it as an except-when or Note directly in the Target Size SC.

**Comment by jwlilly** [comment]
Perhaps I'm misinterpreting the goal of this issue. I thought it was intended translate 2.5.8 to apply to mobile devices and how we should direct readers to interpret the original meaning of the SC. If that's the goal then I think translating how the CSS pixel units apply to mobile devices is sufficient. Directing users a document with other sizes depending whether it is a mouse/cursor input vs touch muddies the water in my mind. 

That being said, I am coming in at the tail-end of this discussion so I may not have the full context. I'll wait for input from others.

**Comment by TanyaVanWorkum** [comment]
@jwlilly  thanks for the React Native testing. We also have a related discussion on units in [#157](https://github.com/w3c/matf/issues/157).

Some additional context here. WCAG2Mobile is itself a W3C Note, and is not normative (and never will be) — only WCAG 2.2 is normative. Like WCAG2ICT, the job of WCAG2Mobile note is to help readers understand how to apply the existing WCAG 2.2 success criteria in a mobile context. 

For 2.5.8, most of the past two years has gone into two questions:

1. How to express "CSS pixels" for mobile without narrowing the unit to one platform. The document is meant to stay technology-agnostic, even though in practice most of our discussion centers on Android and iOS.
2. How to convey that, while the 24×24 requirement maps across cleanly, that size sits at the small end for touch - which is the dominant input mode on the devices we're scoping.

The constraint we've worked under is the one you raised: as an interpretive Note, we cannot introduce numbers or thresholds that aren't in WCAG 2.2. So the changes to the interpretive text stay minimal.

The additional research and recommended touch sizes go into a separate, non-normative Note. That document is a supplementary source - consolidated research and best practices. It's also the right place to translate DIP into real examples like dp and pt, which would help developers. It's deliberately written with a wider audience in mind too: other groups, including work toward WCAG 3.0, may draw on it, so keeping the research in one place has value beyond our own document.

So I think we're aligned: the text in WCAG2Mobile Note stays close to WCAG 2.2, and the separate Note is where the touch-specific research goes, clearly labelled as best practice for mobile.

**Comment by detlevhfischer** [comment]
Just noting that the WCAG normative text does not require a minimum target size, just the minimum clear space to any next target. 
So even if we said targets need to be sized 24x24px we would go beyond the normative text.

**Comment by TanyaVanWorkum** [comment]
[Discussed in MATF meeting on 1 July 2026](https://www.w3.org/2026/07/01-matf-minutes.html)

<details>
<summary>Minutes</summary>

~~~text
shoobe01 No mostly done and looking for feedback - discussion of pixel sizes needs some re-working, I don't like dip. I need to double check other OS's to make sure we get everything uniform. I was trying to avoid being OS specific.
<pauljadam> can you just say 24 x 24 and leave out the "px" ?
shoobe01 I can put in specifics, but I was avoiding that
I wanted to answer pauljadam's question - you can identify the pixel units, but things get very tiny. I think excluding the unit is a bad idea - we should give developers a reference to make it easy for them.
We could always give them a pdf to print and let them cut it out
shoobe01 do we need be explicit on certain sizes
jwlilly - we should include a platform specific reference for them in terms they understand
<pauljadam> in code on iOS you just write midWidth: 24, minHeight: 24 so you don't add pt or px
pauljadam lucky
shoobe01 I forgot to point out - what terms are we using for referring to this - MATF Group Note Draft? Is MATF right? If you don't know we need to keep it in mind
shoobe01 should MATF be in the wording or WCAG's term for mobile?

~~~
</details>

**ACTION:** [review Note 3 text and place your comments regarding the wording ](https://github.com/w3c/matf/issues/10#issuecomment-4800290299)
**ACTION:** review additional [Draft Note on target sizes](https://docs.google.com/document/d/1qbmifBKu_vTXOxbkj6gKOHwBsf-7NVtni0HUkFOgO6k/edit?tab=t.0#heading=h.cos5gmam7jdr) and leave your comments
